### PR TITLE
chore: check ASN syntax in CI; generate ASN docs

### PR DESCRIPTION
**Pull request by sentientwaffle**
Fixes https://github.com/interledger/rfcs/issues/216

### CODE REVIEW

File: scripts/check-asn.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+cat ./asn1/*.asn |
+	# Convert POST body to CRLF
+	sed $'s/$/\r/' |
+	# "@-" means input is from stdin.
+	# "%24" is a URL-encoded "$" (curl expects the field name already escaped).
+	curl --silent -X POST \
+		--data-urlencode '__EVENTTARGET=' \
+		--data-urlencode '__EVENTARGUMENT=' \
+		--data-urlencode '__LASTFOCUS=' \
+		--data-urlencode '__VIEWSTATE=/wEPDwUJOTIzNTU1MjUxD2QWAmYPZBYCAgMPZBYCAgEPZBYKAgEPEGQPFgkCAgIDAgQCBQIGAgcCCAIJAgoWCRAFDklUUyBDQU0gdjEuMy4yBQ5JVFMgQ0FNIHYxLjMuMmcQBQ9JVFMgREVOTSB2MS4yLjIFD0lUUyBERU5NIHYxLjIuMmcQBQlJVFMgSjI3MzUFCUlUUyBKMjczNWcQBQtOQkFQIHY2LjIuMAULTkJBUCB2Ni4yLjBnEAUMUkFOQVAgVjYuMi4wBQxSQU5BUCBWNi4yLjBnEAUKUlJDIHY4LjYuMAUKUlJDIHY4LjYuMGcQBQxTMUFQIHY4LjEwLjAFDFMxQVAgdjguMTAuMGcQBQhUQVAtMDMxMAUIVEFQLTAzMTBnEAULWDJBUCB2OC45LjAFC1gyQVAgdjguOS4wZxYBZmQCDQ8QZGQWAWZkAiEPEGRkFgFmZAIjD2QWAgIBDxBkZBYBZmQCKQ8PFgIeBFRleHQFWnsNCiAgIm5hbWUiOiJBZGFtIiwNCiAgImZpcnN0LXdvcmRzIjoiSGVsbG8gV29ybGQiLA0KICAiYWdlIjp7DQogICAgImJpYmxpY2FsIjo5MzANCiAgfQ0KfWRkGAEFHl9fQ29udHJvbHNSZXF1aXJlUG9zdEJhY2tLZXlfXxYBBR1jdGwwMCRNYWluQ29udGVudCRjYk5vUmVsYXhlZIkssErSHkM+Q9I95FYQz7JtxQKe+/7KUwMgkm90gRoO' \
```
Comment (adrianhopebailie) [inline scripts/check-asn.sh]: Will this viewstate work indefinitely? Might need a step to fetch a fresh one each time.

File: scripts/check-asn.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		--data-urlencode '__VIEWSTATEGENERATOR=CA0B0334' \
+		--data-urlencode '__EVENTVALIDATION=/wEdACNu03QryRSDPrjfaTiZypOg1Cn+EpYSvY49XXrT29VY4DlKzLXE7UdGV+Sp4JynSVOUQai7x7H8epe7LjGkdO1cgsEkCnvatlsBmoGzDwbENG99/MJQf84nGOuvH0Nqme50SlF0/MXZPvJGTJ0aiVXFjEbYeNpryczPOfAPjZw2TmCy9br2E6vqtwSo6KnqMCHIg3IlVtr0nSnHI6hlNOXtzxTmDO+Ma32VfVY1O54Wy0hi53xAgYeVolslSp4aO/uZQKCK8xCNszfebGf8YBtrE5yR22Zoj/1cYhs6e19KIZKKpUt+2UttFP/gcQ0eOF+UdPim13mnFKHrBb176tgjv459Su9vLXbk7rGI/PHmqd2HCK7GUrcrWVpACKFv2zCj/gTCNXHIux3sexLf2Mopi4gRuE0nuD6kWraOU4UYoGvORuieq817u1hpS2zmlXhRoegCA2sek8Eg96dPkbe7PDNaWsaFKb9VSHutwb/KaCXkW74IirQHSJdIWFDq3cm2kYKX8ahKano1pLMtFiSPxK4dCT8Gousc9C6cEqt4teHXznsvma/H84VTT2zlxs7KMKfMA8Uw4bUbj8ELhLnGEKI50kclRu6PN0VxRCywlr3nAyrPdJpffmK+CnokAsZykxZnu/wcnYaPirCH0PVf1jW/aqKFARsnFpzalprHQj2IOx8SU748tw3+7aHsqbaZirrueKudOfnVoNXkJc0LWv1tPfz/mVZH8b+iflS2JW2qCjX/fw53JtwzCwHjR6YfDQnqCSqClXwQKwZNKZ8j' \
```
Comment (adrianhopebailie) [inline scripts/check-asn.sh]: EVENTVALIDATION above with viewstate.

File: scripts/check-asn.sh (hunk 1 above)
Comment (sentientwaffle) [inline scripts/check-asn.sh]: I checked the page with an incognito tab and these variables don't change. I'll leave them for now and add another step if the tests start failing b/c of it.

Comment (michielbdejong) [inline scripts/check-asn.sh]: does it rely on the files being concatenated in a specific order? Or does asn1 allow forward references?

Ah, I guess http://www.oss.com/asn1/products/documentation/asn1-c/asn1c-compiler-options-reference.html#sort implies forward references are OK in asn1.

Comment (sentientwaffle) [inline scripts/check-asn.sh]: Ya, the order that the files are concatenated in doesn't seem to matter.

**Pull request merged**